base_layer/core/src/transactions/transaction/transaction_kernel.rs

```
@@ -61,7 +66,131 @@ pub struct TransactionKernel {
+                            }
+                            fee = Some(map.next_value()?);
+                        },
+                        Field::LockHeight => {
+                            if lock_height.is_some() {
+                                return Err(de::Error::duplicate_field("lock_height"));
+                            }
+                            lock_height = Some(map.next_value()?);
+                        },
+                        Field::Excess => {
+                            if excess.is_some() {
+                                return Err(de::Error::duplicate_field("excess"));
+                            }
+                            excess = Some(map.next_value()?);
+                        },
+                        Field::ExcessSig => {
+                            if excess_sig.is_some() {
+                                return Err(de::Error::duplicate_field("excess_sig"));
+                            }
+                            excess_sig = Some(map.next_value()?);
+                        },
+                    }
+                }
+                let version: u8 = version.ok_or_else(|| de::Error::missing_field("version"))?;
+                match version {
+                    0 => {
+                        let features = features.ok_or_else(|| de::Error::missing_field("features"))?;
+                        let fee = fee.ok_or_else(|| de::Error::missing_field("fee"))?;
+                        let lock_height = lock_height.ok_or_else(|| de::Error::missing_field("lock_height"))?;
+                        let excess = excess.ok_or_else(|| de::Error::missing_field("excess"))?;
+                        let excess_sig = excess_sig.ok_or_else(|| de::Error::missing_field("excess_sig"))?;
+                        Ok(TransactionKernel::new(features, fee, lock_height, excess, excess_sig))
+                    },
+                    _ => unimplemented!("new version needs implementing!"),
```

You'll need to return an error here (otherwise a peer can send an unsupported version and crash the node)